Lib/dataclasses.py

```diff
@@ -1075,7 +1075,9 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
         cmp_fields = (field for field in field_list if field.compare)
         terms = [f'self.{field.name}==other.{field.name}' for field in cmp_fields]
         field_comparisons = ' and '.join(terms) or 'True'
-        body =  [f'if other.__class__ is self.__class__:',
+        body =  [f'if id(self) == id(other):',
```

Comment:
To avoid two function calls, use the `is` operator:
```suggestion
        body =  [f'if self is other:',
```

Reply:
Thanks for the comment, I've changed the code style.